tests/scene/test_curve_3d.h

```diff
@@ -0,0 +1,261 @@
+		CHECK(curve->sample_baked_tilt(curve->get_closest_offset(Vector3(0, 50, 0))) == 0);
+	}
+
+	SUBCASE("sample_baked_up_vector, p_apply_tilt = false") {
+		CHECK(curve->sample_baked_up_vector(curve->get_closest_offset(Vector3(0, 0, 0))) == Vector3(1, 0, 0));
+		CHECK(curve->sample_baked_up_vector(curve->get_closest_offset(Vector3(0, 25, 0))) == Vector3(1, 0, 0));
+		CHECK(curve->sample_baked_up_vector(curve->get_closest_offset(Vector3(0, 50, 0))) == Vector3(1, 0, 0));
+	}
+
+	SUBCASE("sample_baked_up_vector, p_apply_tilt = true") {
+		CHECK(curve->sample_baked_up_vector(curve->get_closest_offset(Vector3(0, 0, 0)), true) == Vector3(1, 0, 0));
+		CHECK(curve->sample_baked_up_vector(curve->get_closest_offset(Vector3(0, 25, 0)), true) == Vector3(1, 0, 0));
+		CHECK(curve->sample_baked_up_vector(curve->get_closest_offset(Vector3(0, 50, 0)), true) == Vector3(1, 0, 0));
+	}
+
+	SUBCASE("get_closest_point") {
+		CHECK(curve->get_closest_point(Vector3(0, 0, 0)) == Vector3(0, 0, 0));
+		CHECK(curve->get_closest_point(Vector3(0, 25, 0)) == Vector3(0, 25, 0));
+		CHECK(curve->get_closest_point(Vector3(50, 25, 0)) == Vector3(0, 25, 0));
+		CHECK(curve->get_closest_point(Vector3(0, 50, 0)) == Vector3(0, 50, 0));
+		CHECK(curve->get_closest_point(Vector3(50, 50, 0)) == Vector3(0, 50, 0));
+		CHECK(curve->get_closest_point(Vector3(0, 100, 0)) == Vector3(0, 50, 0));
+	}
+}
+
+TEST_CASE("[Curve3D] Tessellation") {
+	Ref<Curve3D> curve = memnew(Curve3D);
+	add_sample_curve_points(curve);
+
+	const int default_size = curve->tessellate().size();
+
+	SUBCASE("Increase to max stages should increase num of points") {
+		CHECK(curve->tessellate(6).size() > default_size);
+	}
+
+	SUBCASE("Decrease to max stages should decrease num of points") {
+		CHECK(curve->tessellate(4).size() < default_size);
+	}
+
+	SUBCASE("Increase to tolerance should decrease num of points") {
+		CHECK(curve->tessellate(5, 5).size() < default_size);
+	}
+
+	SUBCASE("Decrease to tolerance should increase num of points") {
+		CHECK(curve->tessellate(5, 3).size() > default_size);
+	}
+
+	SUBCASE("Adding a straight segment should only add the last point to tessellate return array") {
+		curve->add_point(Vector3(0, 100, 0));
+		PackedVector3Array tes = curve->tessellate();
+		CHECK(tes.size() == default_size + 1);
+		CHECK(tes[tes.size() - 1] == Vector3(0, 100, 0));
+		CHECK(tes[tes.size() - 2] == Vector3(0, 50, 0));
+	}
+}
+
+TEST_CASE("[Curve3D] Even length tessellation") {
+	Ref<Curve3D> curve = memnew(Curve3D);
+	add_sample_curve_points(curve);
+
+	const int default_size = curve->tessellate_even_length().size();
+
+	// Default tessellate_even_length tolerance_length is 20.0, by adding a 100 units
+	// straight, we expect the total size to be increased by more than 5,
+	// that is, the algo will pick a length < 20.0 and will divide the straight as
+	// well as the curve as opposed to tessellate() which only adds the final point
```

Comment:
```suggestion
	// well as the curve as opposed to tessellate() which only adds the final point.
```
Just a small nitpick.
Maybe split up the sentence since it's quite long and hard to follow.